src/components/star-rating.tsx

```
@@ -0,0 +1,40 @@
+  iconClassName,
+  text,
+}: Props) => {
+  // Normalise first to ensure we always end up with a finite number
+  const normalised = Number.isFinite(rating) ? rating : MIN_RATING;
+  const safeRating = Math.max(MIN_RATING, Math.min(MAX_RATING, normalised));
+
+  return (
+    <div className={cn("flex items-center gap-x-1", className)}>
+      {Array.from({ length: 5 }, (_, index) => (
+        <StarIcon
+          key={index}
```

Biome (2.1.2)

[error] Avoid using the index of an array as key property in an element.

This is the source of the key value.

The order of the items may change, and this also affects performances and component state.
Check the React documentation. 

(lint/suspicious/noArrayIndexKey)